src/library/components/RadioGroup.tsx

```diff
@@ -50,32 +65,54 @@ const RadioGroup = ({ legend, errorText = '', helpText, children, ...fieldsetAtt
       setFieldsetError: (text) => {
         setValidationMessage(text);
         setIsInvalid(!!text);
+      },
+      buttonGroup: buttonGroupStyle,
+      setButtonGroup: (buttonGroup: boolean) => {
+        setButtonGroupStyle(buttonGroup);
       }
     });
-  }, [isInvalid, errorTextId, errorText, areRadiosDirty]);
+  }, [isInvalid, errorTextId, errorText, areRadiosDirty, buttonGroupStyle]);
 
   useEffect(() => {
     const idNum = ++radioGroupId;
     setHelpTextId('radioGroupHelpText' + idNum);
     setErrorTextId('radioGroupErrorText' + idNum);
   }, []);
 
+  let radioGroupStyles = {
+    fieldsetClass: 'bsds-fieldset',
+    legendClass: 'bsds-legend',
+    inputErrorClass: 'bsds-input-error',
+    inputHelpText: 'bsds-input-help-text'
+  };
+
+  const buttonGroupStyles = {
+    fieldsetClass: 'bsds-fieldset-button-group',
+    legendClass: 'bsds-legend-button-group',
+    inputErrorClass: 'bsds-input-error',
+    inputHelpText: 'bsds-input-help-text'
+  };
+
+  if (buttonGroup) {
+    radioGroupStyles = { ...buttonGroupStyles };
+  }
+
   return (
     <fieldset
-      className="bsds-fieldset"
+      className={radioGroupStyles.fieldsetClass}
```

Comment:
alternative path to swapping class names would be to chain classes at the fieldset level and then have different declarations as children, e.g.:

.bsds-fieldset { .bsds-legend { legend stuff that's default/generic } }
.bsds-fieldset.bsds-fieldset-button-group { .bsds-legend { legend stuff specific to button group } }